primefaces/src/main/java/org/primefaces/component/selectcheckboxmenu/SelectCheckboxMenuRenderer.java

```diff
@@ -318,19 +335,348 @@ protected void encodeMenuIcon(FacesContext context, SelectCheckboxMenu menu, boo
+        writer.writeAttribute("class", panelStyleClass, null);
+        if (panelStyle != null) {
+            writer.writeAttribute("style", panelStyle, null);
+        }
+        writer.writeAttribute(HTML.ARIA_ROLE, "dialog", null);
+
+        if (menu.isShowHeader()) {
+            encodePanelHeader(context, menu, selectItems);
+        }
+
+        writer.startElement("div", null);
+        writer.writeAttribute("class", SelectCheckboxMenu.ITEMS_WRAPPER_CLASS, null);
+        writer.writeAttribute("style", "max-height:" + maxScrollHeight, null);
+
+        if (!menu.isDynamic()) {
+            encodePanelContent(context, menu, selectItems);
+        }
+
+        writer.endElement("div");
+
+        writer.endElement("div");
+    }
+
+    private void encodePanelHeader(FacesContext context, SelectCheckboxMenu menu, List<SelectItem> selectItems) throws IOException {
+        ResponseWriter writer = context.getResponseWriter();
+        Converter converter = menu.getConverter();
+        Object submittedValues = getSubmittedValues(menu);
+        Object values = getValues(menu);
+        Object valuesArray = (submittedValues != null) ? submittedValues : values;
+
+        writer.startElement("div", null);
+        writer.writeAttribute("class", SelectCheckboxMenu.HEADER_CLASS, null);
+
+        //determine if any of the items is not selected
+        boolean notChecked = selectItems.stream()
+                .flatMap(selectItem -> {
+                    if (selectItem instanceof SelectItemGroup) {
+                        //convert children to stream
+                        final SelectItemGroup selectItemGroup = (SelectItemGroup) selectItem;
+                        return Stream.of(selectItemGroup.getSelectItems());
+                    }
+                    else {
+                        return Stream.of(selectItem);
+                    }
+                })
+                .anyMatch(selectItem -> {
+                    final Object value;
+                    if (submittedValues != null) {
+                        //use submitted string representations of values
+                        value = getOptionAsString(context, menu, converter, selectItem.getValue());
+                    }
+                    else {
+                        //use initial values
+                        value = selectItem.getValue();
+                    }
+                    return !isSelected(context, menu, value, valuesArray, converter);
+                });
+
+        //toggler
+        encodeCheckbox(context, null, false, !notChecked, null, "Select All");
+
+        //filter
+        if (menu.isFilter()) {
+            encodeFilter(context, menu);
+        }
+
+        //closer
+        writer.startElement("a", null);
+        writer.writeAttribute("class", SelectCheckboxMenu.CLOSER_CLASS, null);
+        writer.writeAttribute("href", "#", null);
+        writer.writeAttribute(HTML.ARIA_LABEL, "Close", null);
```

Comment:
https://github.com/primefaces/primefaces/blob/master/primefaces/src/main/resources/org/primefaces/Messages.properties#L19

Reply:
`MessageFactory.getMessage(InputRenderer.ARIA_FILTER)` for example if you look in the code for `MessageFactory`

Reply:
Looks like we have a `primefaces.messages.aria.CLOSE = Close` label: https://github.com/primefaces/primefaces/blob/master/primefaces/src/main/resources/org/primefaces/Messages_en.properties